lib/ecto/migration.ex

```diff
@@ -944,8 +944,7 @@ defmodule Ecto.Migration do
     |> List.flatten()
     |> Enum.map(&to_string(&1))
     |> Enum.map(&String.replace(&1, ~r"[^\w_]", "_"))
-    |> Enum.map(&String.replace_trailing(&1, "_", ""))
-    |> Enum.join("_")
+    |> Enum.map_join("_", &String.replace_trailing(&1, "_", ""))
```

Comment:
You should lift this change to a separate PR and if you are going for efficiency, you might as well combine the previous map functions into one anonymous function.

Reply:
Thanks for the feedback! That's a big improvement over what I did initially.